test/unit/cache/sfCacheDriverTests.class.php

```diff
@@ -18,7 +18,8 @@ static public function launch($t, $cache)
     $t->is($cache->get('test'), $data, '->get() retrieves data form the cache');
     $t->is($cache->has('test'), true, '->has() returns true if the cache exists');
 
-    $t->ok($cache->set('test', $data, -10), '->set() takes a lifetime as its third argument');
+    $t->ok($cache->set('test', $data, 1), '->set() takes a lifetime as its third argument');
+    sleep(2);
```

Comment:
About the test duration concern.

number of sleep * sleep duration * number of class
6 * 2 * 4 = 48 seconds is added to the test suite duration, this not count all the matrix with all PHP versions.

I suggest trying to find a way to mitigate this issue.

Does it be possible to use at most 2 sleep per class test (e.g. 2 * 2 * 4 = 16) ?

The test will need to be refactored, for sure.

I may help if you need.

Reply:
Not sure whether this is worth our time optimising for CI time on a framework that is not use used out there?

I mean, I'm all for optimising code performance for production, but I don't think Travis will be bothered if it runs another minute or 5 and the scenario where a dev is manually testing **all** tests with **all** PHP versions is somewhat limited: in most cases, one is manually running a single unit test that failed on a specific version (or skipping most using the `--only-failed` switch).

Personally I would go for not touching the testing logic here even further and save that time to invest on the [main reason](https://github.com/FriendsOfSymfony1/symfony1/issues/261) we are touching this.

Reply:
> I don't think Travis will be bothered if it runs another minute or 5

Just a reminder, Travis-CI is the past as it became paying.

So it is the responsibility, to dev to run all tests, and we know them does not have much time to work on this.

Okay, let's say it's not required, for now, to invest effort to avoid slowing down the test duration.